### DISCUSSION TITLE
Scala 3

### DISCUSSION THREAD

**Discussion by bblfish** [ScalablyTyped/Converter#359] [General]
Hi,
 
  I just came across the library here yesterday, watched the 2019 [Introducing ScalablyTyped](https://twitter.com/bblfish/status/1442397601920212996) presentation, and succeeded in getting [rdflib](https://github.com/linkeddata/rdflib.js/) compiled with `ScalablyTypedConverterGenSourcePlugin`.  

Reading the doc and watching the presentation, I have been wondering how far you are along in producing scala 3 code? You still use the `.^` method even though Scala 3 has top level methods,... Also there are 3 compilation problems in the generated code which I reported in issue https://github.com/ScalablyTyped/Converter/issues/361 . So I wondered what your goals there were in producing more Scala3 ready translations, if you were mostly done there, or if there was going to be a major change.

Also another reason it would help to know what the goals for Scala 3 are, is that it will help me get an idea as to how much I should invest in working with the generated code right now. IF there is going to be some big improvement later, I should perhaps just do the minimum work now I need.

**Comment by oyvindberg** [comment]
Hey!

I have considered Scala 3 support to be "done" in the sense that it works as well as Scala 2 AFAIK. Though I'm sure there are heaps of small things which should be improved.

There are some bigish encoding changes coming up:
- true literal types
- no longer rewrite union types to inheritance for scala 3
- a small revamp of the module system
 
literal types will cause some migration downstream, while the other two shouldn't really cause much changes, at least on the usage site.

For your question on `^` in particular, it's quite hard to get rid of! I really don't like it, but the problem is that it describes a module itself, and it doesn't have a name at all in javascript. You use it like this:
```javascript
const foo = require('some-lib')
new foo()
```
In this case `foo` is a name which is always provided by the user. 

Any inferred name for this thing is quite likely to cause a name collision.
